python/kserve/kserve/model_server.py

```diff
@@ -203,7 +204,12 @@ async def serve():
                 serversocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
                 serversocket.bind(('0.0.0.0', self.http_port))
                 serversocket.listen(5)
-                multiprocessing.set_start_method('fork')
+                start_method = multiprocessing.get_start_method(allow_none=True)
+                if start_method is None:
+                    logger.info("Setting 'fork' as multiprocessing start method.")
+                    multiprocessing.set_start_method('fork')
+                elif start_method in ['spawn', 'forkserver']:
```

Comment:
Can we add inline comment on why we only support 'fork' and not the other 2 ?

Reply:
here the workers are only meant for uvicorn processes, if it is intended to launch a model with multi-processing then it would not work, for example pytorch does not support fork.
https://pytorch.org/docs/stable/notes/multiprocessing.html#cuda-in-multiprocessing

Reply:
> Can we add inline comment on why we only support 'fork' and not the other 2 ? Rest LGTM

There is already a comment above the changed lines. Because of the way github shows changes, you might have missed it. But looking into [this PR](https://github.com/kserve/kserve/pull/2718) will give you more context though :)